src/impl_constructors.rs

```diff
@@ -457,7 +460,8 @@ where
         // debug check for issues that indicates wrong use of this constructor
         debug_assert!(dimension::can_index_slice(&v, &dim, &strides).is_ok());
         ArrayBase {
-            ptr: nonnull_from_vec_data(&mut v),
+            ptr: nonnull_from_vec_data(&mut v)
+                .offset(offset_from_ptr_to_memory(dim.slice(), strides.slice()).abs()),
```

Comment:
Style wise I would use a minus here instead of `.abs()`. Abs would signal that this can return both -3 and 3 and we're fine with either.

Reply:
That makas sense. We should replace abs with a minus sign.